### ISSUE TITLE
Default skin for smaller screens (smaller than the current 1008x500px)

### ISSUE THREAD

**Issue by timewasternl** [mixxxdj/mixxx#11846]
### Feature Description

As Mixxx can also be built on Raspberry Pi's and it is not very uncommon to have 7" touchscreens attached to those, would it be feasible to have a skin present for smaller screens? Apart from this specific use-case, there might be several more ways of running Mixxx on devices with smaller screens, but the skins supplied by default have pretty high minimum resolutions.

![skin_preview](https://github.com/timewasternl/Pioneered/blob/master/skin_preview.png?raw=true)

A while ago, I have started working on [a skin to mimic Pioneer gear](https://github.com/timewasternl/pioneered) (like the XDJ-XZ) with the goal to run on smaller screens (demo [here](https://www.youtube.com/watch?v=eVXIiBcKync)). I can understand it might be too risky to have this included, as it heavily relies on Pioneer's branding, but it could be used as a base.

In conclusion, what do we want/need in order to have such a skin shipped by default? Are there any skin features that are a must?

**Comment by ronso0** [comment]
A touch-optimized skin for smaller screens?
Yes, please!

Mixxx is currently not touch friendly (small controls, many right-click actions, many options only accessible via the menubar), but I reckon that'll change as soon as QML development gains more momentum. Until then a touch skin would be a good step forward, even more so if you already have a solid base.

IMHO we should not mimic certain hardware but pick the best of the options out there. Furthermore, a touch skin should (like the 'desktop' skins) definitively give access to the default effects interface (effect rack with 2/4 units).

**Comment by timewasternl** [comment]
> A touch-optimized skin for smaller screens? Yes, please!
> 
> Mixxx is currently not touch friendly (small controls, many right-click actions, many options only accessible via the menubar), but I reckon that'll change as soon as QML development gains more momentum. Until then a touch skin would be a good step forward, even more so if you already have a solid base.
> 
> IMHO we should not mimic certain hardware but pick the best of the options out there. Furthermore, a touch skin should (like the 'desktop' skins) definitively give access to the default effects interface (effect rack with 2/4 units).

We should compile a list in this issue on the features that would be essential for such skin to have. For the Pioneered, I had a pretty solid example to model the skin after, but for a more versatile skin, we might want different features.

I don't think it's feasible to have every small feature accessible though, as it would easily clutter the interface and/or are not even possible for the current 'skin engine' (awaiting QML support indeed).

Some easy quostions for me to get started:
1. What do we think of the tabbed view with buttons at the top? What tabs do we want?
2. What track metadata is seen as essential? I find album art taking up a lot of space for example.
3. The browser section is rather hacky now. Should the folder selection always be visible, in cost of horizontal space?
4. Four deck support?
5. The effect units are quite extensive for smaller displays to all be there (three/four knobs per deck). Should they be next to the waveforms?
6. Should we make the skin very configurable or rather a "one size fits all" style (where other preferences needs may be found in other skins).

**Comment by ronso0** [comment]
I think the 'tab' approach will work (WStackedWidget), with every 'feature' tab providing a mini deck view at the top.
Features would be Library (full view), Effects, Sampler & Mic/Aux. I think it's also easy to add 'sticky feature view' below the main view, like a condensed view of Effects (bit like the current, collapsed Fx units), Samplers and Mic/Aux (just Talk/Play buttons).

I wonder if this is indeed about the minimal display size or simply about the (font) scaling. IIUC this is primarily about the tocu accessibility, hence such a skin should work on a variety of displays: oldschool 11" 1024x768, 7" FullHD and everything in between. To make that work we could make the skin font size selectable via color schemes.

> The browser section is rather hacky now. Should the folder selection always be visible, in cost of horizontal space?

I think the tracks view should always be visible, + a toggle for the tree view (adjustable width).

FWIW take a look at #11526. This would extremly helpful for fullscreen mode on small devices. For touch devices (no Alt key) we may add a menu toggle to the tab row. This would not be a perfect but a quick solution that would be oakyish IMHO until we switch to QML and rework the entire menu (bar) thing.

> Should we make the skin very configurable or rather a "one size fits all" style (where other preferences needs may be found in other skins).

Assuming such a mini skin would be a controller extension we don't need to cram every option inton it, for example we could omit the rate controls (+- up/down buttons and maybe even the slider). Some cover art / spinny options though would be appreciated IMO.
